specification/network/resource-manager/Microsoft.Network/stable/2019-06-01/virtualWan.json

```diff
@@ -2593,7 +2901,330 @@
       ],
       "description": "List of Vpn-Sites."
     },
-    "VirtualHubProperties": {
+    "VpnServerConfigVpnClientRootCertificatePropertiesFormat": {
+      "properties": {
+        "publicCertData": {
+          "type": "string",
+          "description": "The certificate public data."
+        },
+        "provisioningState": {
+          "readOnly": true,
+          "type": "string",
+          "description": "The provisioning state of the VpnServerConfiguration VPN client root certificate resource. Possible values are: 'Updating', 'Deleting', and 'Failed'."
+        }
+      },
+      "required": [
+        "publicCertData"
+      ],
+      "description": "Properties of VPN client root certificate of VpnServerConfiguration."
+    },
+    "VpnServerConfigVpnClientRootCertificate": {
```

Comment:
I'm not entirely sure why these certificates (which are properties of the vpnServerConfiguration resource type), are partially modeled as if they are an ARM resource. That is, they have an object schema like:
`"id":<>,
"name":<>,
"properties":{}
`
yet they aren't really ARM resources themselves. Do they need to be modeled this way? Would it caused confusion that the `id` isn't actually routable via ARM?

Reply:
All 4 types of certificates which are properties of vpnServerConfiguration resource are child resource of it, so they will have all the schema like id, name, etc. Also, currently we have not exposed any rest apis to directly update these child resources but that's doable.